src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceMarginTaggerProvider.cs

```diff
@@ -56,7 +56,8 @@ protected override ITaggerEventSource CreateEventSource(ITextView textViewOpt, I
                 TaggerEventSources.OnWorkspaceChanged(subjectBuffer, AsyncListener),
                 TaggerEventSources.OnViewSpanChanged(ThreadingContext, textViewOpt),
                 TaggerEventSources.OnDocumentActiveContextChanged(subjectBuffer),
-                TaggerEventSources.OnOptionChanged(subjectBuffer, FeatureOnOffOptions.ShowInheritanceMargin));
+                TaggerEventSources.OnOptionChanged(subjectBuffer, FeatureOnOffOptions.ShowInheritanceMargin),
+                TaggerEventSources.OnOptionChanged(subjectBuffer, FeatureOnOffOptions.InheritanceMarginCombinedWithIndicatorMargin));
```

Comment:
This is the core of the changes.
By subscribing to this option, recompute the tags, and this will force the editor to 'refresh' the glyphs in the indicator margin. And we can use our option service in the glyph factory to just return null.

In our view margin, we do a similar thing, before adding the tag, check if the glyph should be placed in the Indicator margin, if yes, then just clear everything and collapse the canvas.

This will bring a small perf cost (recompute the tag in the document). But since user won't frequently change the option, so I feel this is fine enough

Reply:
> This will bring a small perf cost (recompute the tag in the document, just like changing a character in the document). But since user won't frequently change the option, so I feel this is fine enough

Yeah, this is a _totally_ fine approach to perf.

Reply:
agreed.  this is the ideal case.